acvm-repo/acvm/src/pwg/mod.rs

```diff
@@ -301,6 +295,39 @@ impl<'backend, B: BlackBoxFunctionSolver> ACVM<'backend, B> {
             }
         }
     }
+
+    fn solve_brillig_opcode(
+        &mut self,
+    ) -> Result<Option<ForeignCallWaitInfo>, OpcodeResolutionError> {
+        let Opcode::Brillig(brillig) = &self.opcodes[self.instruction_pointer] else {
+            unreachable!("Not executing a Brillig opcode");
+        };
+        let witness = &mut self.witness_map;
+        if BrilligSolver::<B>::should_skip(witness, brillig)? {
+            BrilligSolver::<B>::zero_out_brillig_outputs(witness, brillig).map(|_| None)
+        } else {
+            let mut solver = match self.brillig_solver.take() {
+                None => {
+                    BrilligSolver::new(witness, brillig, self.backend, self.instruction_pointer)?
+                }
+                Some(solver) => solver,
+            };
```

Comment:
```suggestion
            // If we're resuming execution after resolving a foreign call then
            // there will be a cached `BrilligSolver` to avoid recomputation.
            let mut solver: BrilligSolver<'_, B> = match self.brillig_solver.take() {
                Some(solver) => solver,
                None => {
                    BrilligSolver::new(witness, brillig, self.backend, self.instruction_pointer)?
                }
            };
```

Let's add a note on why we sometimes have this solver. It's a matter of taste but it would be nice to have the `Some` branch first as well.